ttnn/cpp/ttnn/operations/matmul/device/matmul_op_multi_core_reuse_mcast_1d_program_factory.cpp

```diff
@@ -143,12 +154,17 @@ operation::ProgramWithCallbacks create_program_mcast_in0(
 
     uint32_t in1_CB_size = in1_CB_tiles * in1_single_tile_size;
 
-    uint32_t out_block_tiles = per_core_M * per_core_N;
+    uint32_t out_block_tiles = out_block_h * out_block_w;
+    uint32_t out_shard_tiles = per_core_M * per_core_N;
     uint32_t out_CB_tiles = out_block_tiles;  // No double buffer
+    if (output_is_sharded) {
+        out_CB_tiles = out_shard_tiles;
```

Comment:
If the output is sharded, is there any way to avoid having to use per_core_M * per_core_N out CB tiles?

Reply:
Talked to @yugaoTT 
Can't go below shard height and width.